src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1932,8 +1821,25 @@ private static TypeSymbolWithAnnotations InferResultNullability(Conversion conve
                     var methodOpt = conversion.Method;
                     if ((object)methodOpt != null && methodOpt.ParameterCount == 1)
                     {
+                        if (Binder.RequiresLiftedUserDefinedConversion(methodOpt, operandType.TypeSymbol))
+                        {
+                            // PROTOTYPE(NullableReferenceTypes): Conversions: Lifted operator
+                            break;
+                        }
                         // PROTOTYPE(NullableReferenceTypes): Update method based on operandType.
-                        return methodOpt.ReturnType;
+                        var resultType = methodOpt.ReturnType;
+                        // Ensure converted type matches expected ToType, specifically for lifted
+                        // conversions. See LocalRewriter.MakeConversionNode for similar handling.
+                        if (!fromConversionNode)
+                        {
+                            var toType = conversion.BestUserDefinedConversionAnalysis.ToType;
```

Comment:
>toType [](start = 32, length = 6)

How is toType different from targetType?

Reply:
It looks like `toType` can differ from `targetType` by tuple element names at least. (`CodeGenTupleTests.NullCoalesingOperatorWithTupleNames` contains an example where `methodOpt.ReturnType is `(int d1, int d2)`, `toType` is (int a, int c)?` and `targetType` is (int d1, int d2)?`). Should `InferResultNullability` simply return `TypeSymbolWithAnnotations(targetType)` if `methodOpt.ReturnType != targetType`?